src/main.tsx

```diff
@@ -2,13 +2,16 @@ import { StrictMode } from 'react';
 import { createRoot } from 'react-dom/client';
 import { Provider } from 'react-redux';
 
+import { LuaApiProvider } from './api/LuaApiProvider';
 import { store } from './redux/store';
 import App from './App.tsx';
 
 createRoot(document.getElementById('root')!).render(
-  <StrictMode>
-    <Provider store={store}>
-      <App />
-    </Provider>
-  </StrictMode>
+  <Provider store={store}>
+    <LuaApiProvider>
+      <StrictMode>
```

Comment:
Yes, the strict mode runs everything twice, so effects are run & finished twice, which happens way too quickly for a websocket to establish a connection. That has the effect that the websocket isn't properly opened before it should be closed, and then we cant close it properly. So it should not live inside of the strict mode. For the future it would be better to move all websockets into the RTK Query which will handle all that for us but for now we can have them outside of strict mode